### PR TITLE
feat(unify): unsupported browsers

### PR DESCRIPTION
**Pull request by mapsandapps**
- Closes [UNIFY-583]

* Displays a default browser icon if there's not one in our list
* Displays a message in the main browser list if the browser version is unsupported
* Displays a message in the dropdown if the browser version is unsupported

Screen Shot 2022-02-03 at 9 57 41 AM

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

[UNIFY-583]: https://cypress-io.atlassian.net/browse/UNIFY-583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export const Browser = objectType({
   definition (t) {
     t.nonNull.string('channel')
     t.nonNull.boolean('disabled', {
-      resolve: () => false,
+      resolve: (source, args, ctx) => Boolean(!ctx.browser.isVersionSupported(source)),
```
Comment (ZachJW34) [inline packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts:11]: We have `disabled` and `isVersionSupported`, do we need both? I feel like we are missing some functionality like detecting browsers we know won't work (like safari), and in that case the disabled would make sense for browsers like that.

I think we should revert the change to `disabled` for now rather than having two properties that are the inverse of each other.

File: packages/frontend-shared/cypress/support/mock-graphql/longBrowsersList.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ export const longBrowsersList = [
     majorVersion: '73',
     info: 'Info about electron browser',
     isFocusSupported: true,
+    isVersionSupported: true,
```
Comment (ZachJW34) [inline packages/frontend-shared/cypress/support/mock-graphql/longBrowsersList.ts:13]: Let's add a browser that has `isVersionSupported: false` and a `warning` message, and then add a test that these elements show up in the tooltip for `OpenBrowserList.cy.tsx`. `cy.percySnapshot()` would be good to capture this state as well.

File: packages/data-context/src/sources/BrowserDataSource.ts [hunk 3]
```diff
@@ -76,6 +76,14 @@ export class BrowserDataSource {
     return false
   }
 
+  isVersionSupported (obj: FoundBrowser) {
+    return Boolean(!obj.unsupportedVersion)
+  }
+
+  warning (obj: FoundBrowser) {
```
Comment (ZachJW34) [inline packages/data-context/src/sources/BrowserDataSource.ts:83]: We can remove this, it's not being used since the gql-type is inferring this due to the lack of `resolve`

File: packages/frontend-shared/src/gql-components/topnav/TopNav.vue [hunk 4]
```diff
@@ -155,6 +164,24 @@
           <i-cy-circle-check_x24 class="h-24px w-24px icon-dark-jade-100 icon-light-jade-500" />
         </div>
       </template>
+      <template
+        v-else-if="!browser.isVersionSupported"
+        #suffix
+      >
+        <div class="h-16px relative">
```
Comment (ZachJW34) [inline packages/frontend-shared/src/gql-components/topnav/TopNav.vue:171]: Z-index issue?
Screen Shot 2022-02-02 at 5 19 30 PM

File: packages/frontend-shared/src/gql-components/topnav/UnsupportedBrowserTooltip.vue [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      :enter-to-class="`transition-all transform ${placement === 'top' ? 'origin-bottom' : 'origin-top'} scale-y-100`"
+    >
+      <div
+        v-if="scaleUp"
+        class="rounded flex h-fit
+                  bg-gray-900
+                  text-center py-16px px-24px
+                  text-gray-300 text-size-14px leading-20px
+                  absolute content
+                  need-content before:border-solid before:border-transparent before:border-width-8px
```
Comment (ZachJW34) [inline packages/frontend-shared/src/gql-components/topnav/UnsupportedBrowserTooltip.vue:21]: Cool trick: You can group the `before:class-1 before:class2` by wrapping in parenthesis e.g. `before:(class-1 class-2)`. Makes for cleaner templates.

File: packages/frontend-shared/src/gql-components/topnav/UnsupportedBrowserTooltip.vue [hunk 6]
```diff
@@ -0,0 +1,96 @@
+<template>
+  <div
+    ref="wrapper"
```
Comment (ZachJW34) [inline packages/frontend-shared/src/gql-components/topnav/UnsupportedBrowserTooltip.vue:3]: `wrapper` and the ref in the script don't seem to be used.

**Review by ZachJW34 — CHANGES_REQUESTED** [review]

File: packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts (hunk 1 above)
Comment (mapsandapps) [reply packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts:11]: oh, ok. i assumed that since `disabled` was already there and was already set to gray out those browsers in the list that it had some other meaning different than what we're doing here. hence why i left different behavior in the list for `disabled` vs `disabled && !isVersionSupported`. but i'd rather combine them if we won't lose any needed functionality.

File: packages/launchpad/src/setup/OpenBrowserList.vue [hunk 7]
```diff
@@ -18,25 +18,44 @@
...
+            class="top-0 right-0 absolute"
+            placement="top"
+          >
+            <i-cy-circle-bg-question-mark_x16
+              data-cy="unsupported-browser-tooltip-trigger"
+              class="mt-4px mr-8px relative inline-block icon-dark-gray-700 icon-light-gray-200"
+              alt="unsupported browser"
+            />
+            <template #popper>
+              <div>
```
[the remark this answers is not in the document]
Comment (mapsandapps) [reply packages/launchpad/src/setup/OpenBrowserList.vue:40]: The problem i had here is that this text can be quite long. Much longer than shown in the designs. (I was told to use the text we've been using previously, not what's in the designs.) Maybe there's a different text we're already using that's shorter? Not sure, but this was the one I found.

For example:
```
Unsupported Browser
Cypress does not support running Firefox Developer Edition version 52. To use Firefox Developer Edition with Cypress, install a version of Firefox Developer Edition newer than or equal to 78.
```

File: packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts (hunk 1 above)
Comment (mapsandapps) [reply packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts:11]: oh, maybe i misread? you're saying to keep both but put `disabled` back to always being false?

File: packages/launchpad/src/setup/OpenBrowserList.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (ZachJW34) [reply packages/launchpad/src/setup/OpenBrowserList.vue:40]: That's fair then, no change to the width required!

File: packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts (hunk 1 above)
Comment (ZachJW34) [reply packages/graphql/src/schemaTypes/objectTypes/gql-Browser.ts:11]: I was guessing at the real meaning of `disabled` since it doesn't seem to be doing anything. Leaving it as it is requires less changes, but since it's mostly unused I'm ok with reducing it to one property for now if you think that's best.

File: packages/launchpad/src/setup/OpenBrowserList.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (mapsandapps) [reply packages/launchpad/src/setup/OpenBrowserList.vue:40]: cool! i'd love for it to be variable-width, but that seemed kinda tricky and i know we're trying to move fast 🚀

**Review by ZachJW34 — APPROVED** [review]
LGTM!
